src/v/cluster/partition.cc

```diff
@@ -573,7 +573,8 @@ ss::future<> partition::stop() {
         // `partition_manager::do_shutdown` (caller of stop) will assert
         // out on any thrown exceptions. Hence, acquire the units without
         // a timeout or abort source.
-        auto archiver_reset_guard = ss::get_units(_archiver_reset_mutex, 1);
+        auto archiver_reset_guard = co_await ss::get_units(
```

Comment:
We really need a lint check for this 😄

Reply:
haha, true

Reply:
https://github.com/redpanda-data/llvm-project/pull/4/files#diff-4428d32e2826d47ac4cd0a5c3b52d0e146e97186f7d09758b7528748a934820c